piptools/scripts/compile.py

```diff
@@ -381,6 +381,12 @@ def cli(
         if isinstance(output_file, LazyFile):  # pragma: no cover
             ctx.call_on_close(safecall(output_file.close_intelligently))
 
+    for src_file in src_files:
+        if src_file != "-" and src_file == output_file.name:
```

Comment:
Did you mean this?
```suggestion
    if output_file.name != "-" and output_file.name in src_files:
```

Reply:
Neat! Addressed in d7450fc26e7cca462a6f8807d30a9ee52284f08d